### PR TITLE
img: init 0.5.11

### PR DESCRIPTION
**Pull request by superherointj**
img: init 0.5.11

Co-authored-by: @jnetod @thiagokokada

* Tests are failing: `undefined: Asset`
  * Will tests need to be wrapped for runc (as install did)?

### CODE REVIEW

File: pkgs/development/tools/img/default.nix [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    rev = "v${version}";
+    sha256 = "0r5hihzp2679ki9hr3p0f085rafy2hc8kpkdhnd4m5k4iibqib08";
+  };
+
+  vendorSha256 = null;
+
+  postPatch = ''
+    substituteInPlace ./internal/unshare/unshare.c \
+      --replace "/usr/bin/newgidmap" "/run/wrappers/bin/newgidmap" \
+      --replace "/usr/bin/newuidmap" "/run/wrappers/bin/newuidmap"
```
Comment (thiagokokada) [inline pkgs/development/tools/img/default.nix:24]: Shouldn't this point to something like `{package}/bin/{newgidmap,newuidmap}` instead :thinking: ?

Comment (thiagokokada) [reply pkgs/development/tools/img/default.nix:24]: Like here: https://github.com/NixOS/nixpkgs/blob/master/pkgs/tools/security/nsjail/default.nix#L27

Comment (thiagokokada) [reply pkgs/development/tools/img/default.nix:24]: If for some reason you really need to use binaries from `/run`, I think your package will only work on NixOS.

Comment (superherointj) [reply pkgs/development/tools/img/default.nix]: Thanks for your insightful input and assistance. I have parameterized the wrapper directory.

File: pkgs/development/tools/img/default.nix [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        --replace "/usr/bin/$V" "${wrapperDir}/$V"
+  '';
+
+  nativeBuildInputs = [
+    makeWrapper
+  ];
+
+  propagatedBuildInputs = [
+    runc
+  ];
```
Comment (SuperSandro2000) [inline pkgs/development/tools/img/default.nix:34]: This shouldn't be required with makeWrapper.

Comment (superherointj) [reply pkgs/development/tools/img/default.nix]: Done

File: pkgs/development/tools/img/default.nix [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    rev = "v${version}";
+    sha256 = "0r5hihzp2679ki9hr3p0f085rafy2hc8kpkdhnd4m5k4iibqib08";
+  };
+
+  vendorSha256 = null;
+
+  postPatch = ''
+    V={newgidmap,newgidmap} \
+      substituteInPlace ./internal/unshare/unshare.c \
+        --replace "/usr/bin/$V" "${wrapperDir}/$V"
```
Comment (SuperSandro2000) [inline pkgs/development/tools/img/default.nix:25]: https://github.com/genuinetools/img/blob/master/internal/unshare/unshare.c#L134 for reference

Comment (superherointj) [reply pkgs/development/tools/img/default.nix:25]: What do you mean?

**Comment by vcunat** [comment]
Fun fact: this broke stuff (not your fault really)
https://github.com/NixOS/nixpkgs/pull/140063

**Pull request merged**